plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java

```diff
@@ -354,6 +354,12 @@ void fail(Throwable e)
             ImmutableList.Builder<ConnectorSplit> resultBuilder = ImmutableList.builder();
             int removedEstimatedSizeInBytes = 0;
             for (InternalHiveSplit internalSplit : internalSplits) {
+                // Dynamic filter may not have been ready when partition was loaded in BackgroundHiveSplitLoader.
```

Comment:
There is a check like that at https://github.com/trinodb/trino/pull/9866/files#diff-0434feb95d650801292ef684f874cea122b5274de0a7f71a0c593da59e701d45R166 too
Dynamic filtering of splits was originally moved out of HiveSplitSource and into createInternalHiveSplit on the basis of https://github.com/trinodb/trino/pull/1072#discussion_r353303931
We can probably remove the extra check from createInternalHiveSplit now as there isn't anything too expensive happening between that and this

Reply:
btw how expensive is the check by itself?

Reply:
There is a small overhead to it but so far i haven't seen it show up on JFR profiles from coordinator